From f7ea534a0920dbaf71a8003936e178e14ec9271d Mon Sep 17 00:00:00 2001
Message-Id: <f7ea534a0920dbaf71a8003936e178e14ec9271d.1659928159.git.geliang.tang@suse.com>
From: Jakub Kicinski <kuba@kernel.org>
Date: Wed, 15 Dec 2021 18:55:36 -0800
Subject: [PATCH] add includes masked by cgroup -> bpf dependency
Git-commit: f7ea534a0920dbaf71a8003936e178e14ec9271d
Patch-mainline: v5.17-rc1
References: bsn#210
Modified-by-SEL: Yes, refreshed due to context changes

cgroup pulls in BPF which pulls in a lot of includes.
We're about to break that chain so fix those who were
depending on it.

Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Link: https://lore.kernel.org/bpf/20211216025538.1649516-2-kuba@kernel.org
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 arch/s390/mm/hugetlbpage.c | 1 +
 include/linux/perf_event.h | 1 +
 2 files changed, 2 insertions(+)

--- a/arch/s390/mm/hugetlbpage.c
+++ b/arch/s390/mm/hugetlbpage.c
@@ -9,6 +9,7 @@
 #define KMSG_COMPONENT "hugetlb"
 #define pr_fmt(fmt) KMSG_COMPONENT ": " fmt
 
+#include <asm/pgalloc.h>
 #include <linux/mm.h>
 #include <linux/hugetlb.h>
 #include <linux/mman.h>
--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -609,6 +609,7 @@ struct swevent_hlist {
 #define PERF_ATTACH_ITRACE	0x10
 #define PERF_ATTACH_SCHED_CB	0x20
 
+struct bpf_prog;
 struct perf_cgroup;
 struct perf_buffer;
 
